### DISCUSSION TITLE
cert does not exist error

### DISCUSSION THREAD

**Discussion by alanlubold** [gravitational/teleport#20034] [Q&A] [answered]
I am setting up version 11 for the first time, and am getting this error: [1025]: ERROR: https cert does not exist:

We want only server access, no application or database access. We have used versions 4 and 5 successfully for some time but need the web_idle_timeout feature. Our implementation will be to have one server (this one) and a number of Trusted Clusters. 

I do not seem to be getting the teleport.yaml configuration right, or perhaps it is my implementation of Letsencrypt. I do not have a wildcard cert in place, and it would seem I don't need that, even though that is the way the reference shows it. My DNS provider is Dreamhost and they do not have a plugin so I'd rather stick with a cert for the server only if I can, i.e., tel11.arctevity-cloud.com not *.tel11.arctevity-cloud.com. There are no subdomains.   

I have attached the configs and output in question as a file to preserve the formatting. 

[cert_error.txt](https://github.com/gravitational/teleport/files/10383967/cert_error.txt)

Thanks,

Alan

**Comment by alanlubold** [comment, marked as answer]
Resolved by changing the format of the teleport.yaml file. 


[new_teleport.yaml.txt](https://github.com/gravitational/teleport/files/10394434/new_teleport.yaml.txt)

**Reply by webvictim** [reply]
FYI the correct format to use in your original config file looks like this:
```diff
proxy_service:
  https_keypairs: 
  - key_file: /etc/letsencrypt/live/tel11.arctevity-cloud.com/privkey.pem
- - cert_file: /etc/letsencrypt/live/tel11.arctevity-cloud.com/fullchain.pem
+   cert_file: /etc/letsencrypt/live/tel11.arctevity-cloud.com/fullchain.pem
```

`https_keypairs` is a list which can have one or more elements under it. Each element must have exactly one `key_file` value and exactly one `cert_file` value. Before you had a `-` at the beginning of both values which means it treats them as separate elements.

You should be able to validate this on the command line using `teleport configure --test /etc/teleport.yaml` too.

**Comment by alanlubold** [comment]
Set the web_idle_timeout value to 12h, but a test session timed out after a few hours. What is the syntax of this value, and what is the limit, if there is one?

**Reply by webvictim** [reply]
The value is correct the way you specified it. If the connection drops for any other reason (say a stateful firewall timeout, or a lack of keepalive traffic on the connection) then the session will also terminate. This can also happen if the browser tab is backgrounded due to power saving logic in the browser, etc. `web_idle_timeout` is an upper limit rather than a guarantee that the session will stay alive for that length of time.

**Comment by alanlubold** [comment]
Thanks for both of these answers.

**Comment by alanlubold** [comment]
New issue has come up on trying to join a leaf cluster to the root cluster I've been working on. 

The specific error is this:

**arc@6-0-0-0-12:/etc $ sudo /usr/local/bin2/tctl create /etc/trusted_cluster.yaml 
ERROR: access denied to perform action "read" on "trusted_cluster"**

(Line 235 in the attached)

I have attached various files to show the config I have on the two machines. My goal here is to add a few more machines to the leaf cluster; they are machines at a remote customer site and the root machine is in the cloud, giving access to the customer machines behind a firewall. 

The root machine has the old-style config where the trusted_cluster value is in the teleport.yaml, I tried it multiple times using this value in the trusted_cluster file on the leaf but got the same error (that's why that value is still in the file on the root). I then went to the ephemeral token version, but had the same error so I thought this wasn't the problem. 

I'm guessing I have something wrong in my roles giving me a permissions problem, but I can't se what it is. 

Let me know if there is any other data I can provide. Thanks.

[trusted_cluster_issue.txt](https://github.com/gravitational/teleport/files/10418437/trusted_cluster_issue.txt)

**Comment by alanlubold** [comment]
Previous post has been moved to a new issue, 20301
